lib/internal/modules/typescript.js

```diff
@@ -63,10 +63,14 @@ function parseTypeScript(source, options) {
      * It allows us to distinguish between invalid syntax and unsupported syntax.
      */
     switch (error?.code) {
-      case 'UnsupportedSyntax':
-        throw new ERR_UNSUPPORTED_TYPESCRIPT_SYNTAX(error.message);
-      case 'InvalidSyntax':
-        throw new ERR_INVALID_TYPESCRIPT_SYNTAX(error.message);
+      case 'UnsupportedSyntax': {
+        const unsupportedSyntaxError = new ERR_UNSUPPORTED_TYPESCRIPT_SYNTAX(error.message);
+        throw decorateErrorWithSnippet(unsupportedSyntaxError, error); /* node-do-not-add-exception-line */
```

Comment:
Could we get please the linter with the following?
```suggestion
        const unsupportedSyntaxError = new ERR_UNSUPPORTED_TYPESCRIPT_SYNTAX(error.message);
        decorateErrorWithSnippet(unsupportedSyntaxError, error);
        throw unsupportedSyntaxError;
```

Reply:
Not sure wdym, the annotation is very important to avoid adding the line of the internal where it was generated. Its like a pre processor thing rather than a linter